### DISCUSSION TITLE
Rounding vs truncating in sensors

### DISCUSSION THREAD

**Discussion by Swallowtail23** [mletenay/home-assistant-goodwe-inverter#294] [General]
Hi all,

I note that sensor.today_s_pv_generation seems to truncate values instead of rounding them - is that a limitation of the data provided by the inverter?

E.g., noting the following, the first two sensors are from the integration or derivative from them; the subsequent ones with 2 decimal places are via Home Assistant template/integration sensors from power (W) sensors (which round to 2 decimal places).

![image](https://github.com/mletenay/home-assistant-goodwe-inverter/assets/8214128/c7e30521-5d81-4d7b-a4ff-b440d282ce9f)

BTW, the GoodWe sensor.today_s_pv_generation sensor rolls to the next value very closely timed to the integration sensors, e.g. from the screenshot it rolled to 5.2 from 5.1 in similar timeframe to the integration sensors rolling from 5.19 to 5.20

**Comment by mletenay** [comment]
There is no rounding or truncating at the integration level, all the data are taken as-is from the inverter.
In other words, yes, it is limitation of the provided data.
